iset.mm

```diff
@@ -105304,15 +105345,21 @@ equinumerous to omega (the set of natural numbers as ordinals).
     fxnn0nninf.i $e |- I = ( ( F o. `' G )
       u. { <. +oo , ( _om X. { 1o } ) >. } ) $.
     $( A function from ` NN0* ` into ` NN+oo ` .  (Contributed by Jim Kingdon,
-       16-Jul-2022.) $)
+       16-Jul-2022.)  TODO: use ~ infnninf instead of ~ infnninfOLD .  More
+       generally, this theorem and most theorems in this section could use an
+       extended ` G ` defined by
+       ` G = ( frec ( ( x e. ZZ |-> ( x + 1 ) ) , 0 ) u. <. _om , +oo >. ) `
```

Comment:
I went to some effort to provide a similar mechanism in https://us.metamath.org/ileuni/df-ihash.html and as far as I've noticed we are only using the finite case there.

As for these theorems relating ` NN0* ` into ` NN+oo `, I do sort of like the way they illustrate the comparisons between those two, but on the other hand, there aren't any proofs that use them, so it makes it even harder than usual to me to know which version better serves their purpose (because it is pedagogical).

I guess I'm saying I could see going either way about whether to follow up on the suggestion in the "could use" comment here.

Reply:
> I went to some effort to provide a similar mechanism in https://us.metamath.org/ileuni/df-ihash.html

This is also what I (would like to) do in https://us.metamath.org/mpeuni/df-bj-iomnn.html.

There is another point: in https://us.metamath.org/ileuni/fxnn0nninf.html and the following three theorems, the function `I : NN0* --> NN+oo` is constructed as follows:
* define  `G = frec ... `
* define ` F = ( n e. _om |-> ... `
* and then, define `` I = ( F o. `' G ) u. { special-couple } ``

that is, we do not know that `I` can be naturally factorized into two natural functions.  Whereas, if we define
* define  ` G = frec ... u. { special-couple' } `
* define ` F = ( n e. suc _om |-> ... `
* and then, define `` I = ( F o. `' G ) ``

then we see that `I` is the composition of two natural functions.  Relatedly, the form of the special couple is simpler in the latter case: ` <. _om , +oo >. ` instead of ` <. +oo , ( _om X. { 1o } ) >. `.

Reply:
Yeah, once we've decided to do things with `NN0*` there isn't much point in getting worried about whether the whole concept doesn't make sense.

Adjoining the special couplet to ` G `, as suggested above, probably is a bit nicer than what we have. The key theorem is https://us.metamath.org/ileuni/inftonninf.html and would be provable either way.